dvc/api.py

```diff
@@ -69,3 +75,108 @@ def _make_repo(repo_url, rev=None):
     else:
         with external_repo(url=repo_url, rev=rev) as repo:
             yield repo
+
+
+def summon(name, fname="dvcsummon.yaml", args=None, repo=None, rev=None):
+    # TODO: Write a meaningful docstring about `summon`
+    with _make_repo(repo, rev=rev) as _repo:
+
+        def pull_dependencies(deps):
```

Comment:
1. Might be moved outside.
2. This is actually `repo.pull(targets)` after granularity patch is in, see https://github.com/iterative/dvc/pull/3002